careers/base/middleware.py

```diff
@@ -0,0 +1,25 @@
+from django.http import HttpResponseRedirect
+
+
+class LocaleRedirectionMiddleware(object):
+    """Remove the /en-US/ locale part from the URL.
+
+    The sugardough based version of the app doesn't not enable the
+    locale middleware for simplicity since the site is not localized.
+    To avoid breaking old links that include /en-US/ locale
+    identification string, this middleware will automatically convert
+    them.
+
+    """
+
+    def process_request(self, request):
+        if not request.path.startswith('/en-US/'):
+            return
+
+        _, locale, url = request.path.split('/', 2)
```

Comment:
Instead of rebuilding the params yourself you should be able to just use `request.get_full_path()` and split the locale off of that. And if it's always "/en-US" just do `return request.get_full_path()[6:]`